gallery/__init__.py

```diff
@@ -180,24 +180,35 @@ def upload_file(auth_dict: Optional[Dict[str, Any]] = None):
             assert dir_path
             assert parent
             assert owner
-            file_model = add_file(filename, dir_path, parent, "", owner)
+            mime, file_model = add_file(filename, dir_path, parent, "", owner)
 
             # Upload File
             file_stat = os.stat(filepath)
             with open(filepath, "rb") as f_hnd:
                 storage_interface.put(
                     "files/{}".format(file_model.s3_id),
-                    f_hnd
+                    f_hnd,
+                    filename,
+                    mime
                 )
             os.remove(filepath)
 
             # Upload Thumbnail
             filepath = os.path.join(dir_path, file_model.thumbnail_uuid)
             file_stat = os.stat(filepath)
             with open(filepath, "rb") as f_hnd:
+                print("AAAA",file_model.thumbnail_uuid)
```

Comment:
Haven't looked at the functionality itself, but you prob want to remove these debug prints lol

Reply:
Oh shoot, didn't mean to actually put these in here, oops